packages/main/src/RangeSlider.js

```diff
@@ -0,0 +1,382 @@
+ *
+ * <h3 class="comment-api-title">Overview</h3>
+ * The purpose of the control is to enable visual selection of sub-ranges within a given interval.
+ *
+ * <h3>Structure</h3>
+ * The most important properties of the Range Slider are:
+ * <ul>
+ * <li>min - The minimum value of the slider range</li>
+ * <li>max - The maximum value of the slider range</li>
+ * <li>value - The current value of the slider</li>
+ * <li>step - Determines the increments in which the slider will move</li>
+ * <li>showTooltip - Determines if a tooltip should be displayed above the handle</li>
+ * <li>showTickmarks - Displays a visual divider between the step values</li>
+ * <li>labelInterval - Labels some or all of the tickmarks with their values.</li>
+ * </ul>
+ * <h4>Notes:<h4>
+ * <ul>
+ * <li>The right and left handle can be moved individually and their positions could therefore switch.</li>
+ * <li>The entire range can be moved along the interval.</li>
+ * </ul>
+ * <h3>Usage</h3>
+ * The most common usecase is to select and move sub-ranges on a continuous numerical scale.
+ *
+ * <h3>Responsive Behavior</h3>
+ * You can move the currently selected range by clicking on it and dragging it along the interval.
+ *
+ * <h3>ES6 Module Import</h3>
+ *
+ * <code>import "@ui5/webcomponents/dist/RangeSlider";</code>
+ *
+ *
+ * @constructor
+ * @author SAP SE
+ * @alias sap.ui.webcomponents.main.RangeSlider
+ * @extends sap.ui.webcomponents.base.UI5Element
+ * @tagname ui5-range-slider
+ * @since 1.0.0-rc.11
+ * @appenddocs SliderBase
+ * @public
+ */
+class RangeSlider extends SliderBase {
+	static get metadata() {
+		return metadata;
+	}
+
+	static get template() {
+		return RangeSliderTemplate;
+	}
+
+	constructor() {
+		super();
+		this._stateStorage.startValue = null;
+		this._stateStorage.endValue = null;
+		this.i18nBundle = getI18nBundle("@ui5/webcomponents");
+	}
+
+	get tooltipStartValue() {
+		const stepPrecision = this.constructor._getDecimalPrecisionOfNumber(this._effectiveStep);
+		return this.startValue.toFixed(stepPrecision);
+	}
+
+	get tooltipEndValue() {
+		const stepPrecision = this.constructor._getDecimalPrecisionOfNumber(this._effectiveStep);
+		return this.endValue.toFixed(stepPrecision);
+	}
+
+	/**
+	 * Check if the previously saved state is outdated. That would mean
+	 * either it is the initial rendering or that a property has been changed
+	 * programatically - because the previous state is always updated in
+	 * the interaction handlers.
+	 *
+	 * Normalize current properties, update the previously stored state.
+	 * Update the visual UI representation of the Slider.
+	 *
+	 */
+	onBeforeRendering() {
+		if (!this.isCurrentStateOutdated()) {
+			return;
+		}
+
+		this.notResized = true;
+		this.syncUIAndState("startValue", "endValue");
+		this._updateHandlesAndRange(null);
+	}
+
+	/**
+	 * Called when the user starts interacting with the slider
+	 *
+	 * @private
+	 */
+	_onmousedown(event) {
```

Comment:
I found the following methods a little hard to read: _onmousedown _handleMove _calculateRangeOffset _setCurrentValueType. Could you try to slice them into logical pieces and smaller methods.

Reply:
I broke them apart except for the _setCurrentValueType method. I think it will only make it harder to follow if I compose it from few smaller functions. But I changed its name and added additional comments to improve its readability.